### ISSUE TITLE
Bug: keymanager API does not return slashing data for a deleted key

### ISSUE THREAD

**Issue by yorickdowne** [status-im/nim-beacon-chain#3939]
**Describe the bug**

On a subsequent `DELETE` call of `/eth/v1/keystores` Nimbus will return `not_found`. It should return `not_active` and the slashing protection data.

From https://github.com/ethereum/keymanager-APIs/blob/master/apis/local_keystores.yaml

"Additionally, DELETE must fetch the slashing protection data for the requested keys from
persistent storage, which must be retained (and not deleted) after the response has been sent. Therefore in the
case of two identical delete requests being made, both will have access to slashing protection data."

**Comment by zah** [comment]
So far, I'm not able to reproduce this. The logic in our API handler looks correct and I always get the `not_active` plus slashing data result in my own testing.

**Comment by yorickdowne** [comment]
I just tested this with a fresh source build from `unstable`

```
yorick@ethlinux:~/eth-docker-devel$ ./ethd keys list
Creating eth-docker-devel_validator-keys_run ... done
Validator public keys
0xb44e54bd9715ef64e6963dd849a406ec54258b72278391ce1a241b2756c4ef2af47f47f9ec9258f6913fbe3ddd9497e1
0xa54bbe88143501ed50f7686967de1cbc158b6f1b121a929adc98fb56fae9a19ee8641b40e4c716f1ba2d631de0271d74
yorick@ethlinux:~/eth-docker-devel$ ./ethd keys delete 0xb44e54bd9715ef64e6963dd849a406ec54258b72278391ce1a241b2756c4ef2af47f47f9ec9258f6913fbe3ddd9497e1
Creating eth-docker-devel_validator-keys_run ... done
Validator deleted.
Slashing protection data written to .eth/validator_keys/slashing_protection-0xb44e54bd--dd9497e1.json
yorick@ethlinux:~/eth-docker-devel$ ./ethd keys delete 0xb44e54bd9715ef64e6963dd849a406ec54258b72278391ce1a241b2756c4ef2af47f47f9ec9258f6913fbe3ddd9497e1
Creating eth-docker-devel_validator-keys_run ... done
The key was not found in the keystore, no slashing protection data returned.
```

This code path is used when `status` is `not_found` instead of `not_active`

Taking a look at the returned object:
```
{"data":[{"status":"not_found","message":""}],"slashing_protection":{"metadata":{"interchange_format_version":"5","genesis_validators_root":"0x043db0d9a83813551ee2f33450d23797757d430911a9320530ad8a0eabc43efb"},"data":[]}}
```

The standard says "Slashing protection data must only be returned for keys from request.pubkeys for which a deleted or not_active status is returned."

I believe Nimbus should, in this case, return `not_active` and the slashing protection data, instead of `not_found` and the slashing protection data.

**Comment by yorickdowne** [comment]
Ping on this: Can you reproduce? In the meantime I have a workaround in place for it.

**Comment by yorickdowne** [comment]
Solved in 22.9.0. Thanks!
